apps/hexdailystats/hex_daily_stats.star

```diff
@@ -0,0 +1,186 @@
+"""
+Applet: HEX Daily Stats
+Summary: HEX Daily Stats
+Description: Displays HEX price, Payout per T-Share and T-Share rate
+Author: kmphua
+Thanks: aschober, bretep, codeakk
+"""
+
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+load("math.star", "math")
+
+COINGECKO_PRICE_URL = "https://api.coingecko.com/api/v3/coins/{}?localization=false&tickers=false&community_data=false&developer_data=false"
+
+HEX_ICON_SM = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAAkAAAAICAYAAAArzdW1AAAACXBIWXMAAAsTAAAL
+EwEAmpwYAAABG0lEQVQYlQXBPUsCYQDA8f+dz6WXr3cqZlpkkSCBBI2NITSI0NTi
+B2jpOzT1CRqipYjmliKiJQiioWhoiYqIuDDMzJcwz7vTp99PkZUNeJw3qNkPuIko
+6hTSmwBftqPkkkV0qyUQEtTbA/yjLNok9L9QlDTSqYelld9VCoF1RVZXy1jdE37N
+bZr+OF7cxUsU+AmvIGdgLr/o26qo10THTmGQR/4to7kxDO0IpzuN3Uxh2yXBgrrD
+Z7uENooRDn7z0V9i4KjMpo55aRWRzpggIw7RB0XG3XuS/SbpyCvP1hpDLuj72hjB
+qiCuvxHqnOM19ghEwBT7xMQZ77UyunKHkbkSjDzImpeojRZtt05v0CMyfMIMpQnl
+Nrlx+AeJZmcOHgyOkAAAAABJRU5ErkJggg==
+""")
+
+NO_DATA = "---------- "
+
+HEX_SUBGRAPH_URL = "https://api.thegraph.com/subgraphs/name/codeakk/hex"
+POST_HEADERS = {
+    "Content-Type": "application/json",
+}
+DAILY_DATA_UPDATES_QUERY = "{\"query\": \"{ dailyDataUpdates(orderBy: timestamp orderDirection: desc first: 1) { payoutPerTShare } }\" }"
+GLOBAL_INFOS_QUERY = "{\"query\": \"{ globalInfos(orderBy: timestamp orderDirection: desc first: 1)  { hexDay, shareRate } }\" }"
+
+def main(config):
+    # Get coin data for selected coin from CoinGecko
+    coin_data = get_json_from_cache_or_http(COINGECKO_PRICE_URL.format("hex"), ttl_seconds = 600)
```

Comment:
I think this will be fine, but wanted to mention that there's some risk we'll end up hitting Coingecko's rate limit for the free API. This will be the third app using Coingecko with a 600s TTL, and I'm guessing it'll fit in the free tier, but if not we'll have to revisit and fold this (and the other) apps in under Tidbyt's paid plan (currently in use for the Crypto app).

Reply:
Coingecko's free tier is rate-limited at 50 calls/min so I think should be ok.